test/object-store/sync/flx_sync.cpp

```diff
@@ -255,6 +264,96 @@ TEST_CASE("flx: query on non-queryable field results in query error message", "[
+    // and are bigger than 1MB respectively.
+    //
+    // So this generates 5 objects each with 1000+ instructions that are each 1MB+ big. This should result in
+    // 3 download messages total with one changeset each for the bootstrap download messages.
+    std::vector<ObjectId> obj_ids_at_end;
+    harness.load_initial_data([&](SharedRealm realm) {
+        CppContext c(realm);
+        for (int i = 0; i < 5; ++i) {
+            auto id = ObjectId::gen();
+            auto obj = Object::create(c, realm, "TopLevel",
+                                      util::Any(AnyDict{{"_id", id},
+                                                        {"list_of_strings", AnyVector{}},
+                                                        {"queryable_int_field", static_cast<int64_t>(i * 5)}}));
+            List str_list(obj, realm->schema().find("TopLevel")->property_for_name("list_of_strings"));
+            for (int j = 0; j < 1024; ++j) {
+                str_list.add(c, util::Any(std::string(1024, 'a' + (j % 26))));
+            }
+
+            obj_ids_at_end.push_back(id);
+        }
+    });
+    SyncTestFile interrupted_realm_config(harness.app()->current_user(), harness.schema(),
+                                          SyncConfig::FLXSyncEnabled{});
+    interrupted_realm_config.cache = false;
+
+    {
+        SharedRealm realm;
+        auto [interrupted_promise, interrupted] = util::make_promise_future<void>();
+        int download_msg_counter = 0;
+        Realm::Config config = interrupted_realm_config;
+        config.sync_config->on_download_message_received_hook =
+            [&realm, &download_msg_counter,
+             promise = std::make_shared<util::Promise<void>>(std::move(interrupted_promise))]() mutable {
+                REALM_ASSERT(realm);
+                // We interrupt on the 5rd download message, which should be 1/3rd of the way through the
+                // bootstrap.
+                if (++download_msg_counter != 5) {
+                    return;
+                }
+
+                realm->sync_session()->close();
+                promise->emplace_value();
+            };
+
+        realm = Realm::get_shared_realm(config);
+        {
+            auto mut_subs = realm->get_latest_subscription_set().make_mutable_copy();
+            auto table = realm->read_group().get_table("class_TopLevel");
+            mut_subs.insert_or_assign(Query(table));
+            std::move(mut_subs).commit();
+        }
+
+        interrupted.get();
+    }
+
+    {
+        auto realm = DB::create(sync::make_client_replication(), interrupted_realm_config.path);
+        auto sub_store = sync::SubscriptionStore::create(realm, [](int64_t) {});
+        REQUIRE(sub_store->get_active_and_latest_versions() == std::pair<int64_t, int64_t>{0, 1});
+        auto latest_subs = sub_store->get_latest();
+        REQUIRE(latest_subs.state() == sync::SubscriptionSet::State::Bootstrapping);
```

Comment:
This section is checking whether we've actually set this test up correctly and that there is a SubscriptionSet that has started bootstrapping but hasn't completed yet.